crypto/elliptic_curves/elliptic_curves.go

```diff
@@ -0,0 +1,67 @@
+package elliptic_curves
+
+import (
+	"crypto/elliptic"
+	"crypto/rand"
+	"math/big"
+)
+
+type ellipticCurve struct {
+	curve elliptic.CurveParams
```

Comment:
I think the eliptic curve should be kept like the bigInt context per execution. It is not needed to create and destroy a new eliptic curve for every operation. 
So the basic usage of a SC is to create EC - make a big set of operations on the same EC - and finish execution. So one / two ECs are created at most ,  but many operations on those.

Reply:
Good idea.